libs/designer-ui/src/lib/editor/base/utils/__test__/helper.spec.ts

```diff
@@ -33,3 +33,49 @@ describe('Helper functions', () => {
     ]);
   });
 });
+
+describe('getDropdownOptionsFromOptions', () => {
+  it('returns options directly when they are available', () => {
+    const editorOptions = {
+      options: [
+        { key: '1', value: 'Option 1', displayName: 'Option 1' },
+        { key: '2', value: 'Option 2', displayName: 'Option 2' },
+      ],
+    };
+    const result = getDropdownOptionsFromOptions(editorOptions);
+    expect(result).toEqual(editorOptions.options);
+  });
+
+  it('returns options when they are nested in an object', () => {
+    const editorOptions = {
+      options: {
+        nested: [
+          { key: '1', value: 'Option 1', displayName: 'Option 1' },
+          { key: '2', value: 'Option 2', displayName: 'Option 2' },
+        ],
+      },
+    };
+    const result = getDropdownOptionsFromOptions(editorOptions);
+    expect(result).toEqual(editorOptions.options.nested);
+  });
+
+  it('handles cases where displayName is not a string', () => {
+    const editorOptions = {
+      options: [
+        { key: '1', value: 'Option 1', displayName: { text: 'Option 1' } },
+        { key: '2', value: 'Option 2', displayName: ['Option 2'] },
+      ],
+    };
+    const result = getDropdownOptionsFromOptions(editorOptions);
+    expect(result).toEqual([
+      { key: '1', value: 'Option 1', displayName: '{"text":"Option 1"}' },
+      { key: '2', value: 'Option 2', displayName: '["Option 2"]' },
+    ]);
+  });
+
+  it('returns an empty array when no options are available', () => {
+    const editorOptions = {};
```

Comment:
again why in the editorOptions would be not have options? this is coming from swagger right... do you see any connector with such info? Even if that is the case in the place where we set editorOPtions we should set it to options: []

Reply:
this was just a safety check in a test